packages/mobile/src/screens/upload-screen/utils/computeLicenseIcons.ts

```diff
@@ -0,0 +1,27 @@
+import type { ComponentType } from 'react'
+
+import type { Nullable } from '@audius/common'
+import type { SvgProps } from 'react-native-svg'
+
+import IconAllowAttribution from 'app/assets/images/creativeCommons/by.svg'
+import IconCreativeCommons from 'app/assets/images/creativeCommons/cc.svg'
+import IconNonCommercialUse from 'app/assets/images/creativeCommons/nc.svg'
+import IconNoDerivatives from 'app/assets/images/creativeCommons/nd.svg'
+import IconShareAlike from 'app/assets/images/creativeCommons/sa.svg'
+
+export const computeLicenseIcons = (
+  allowAttribution: boolean,
+  commercialUse: boolean,
+  derivativeWorks: Nullable<boolean>
+) => {
+  if (!allowAttribution) return null
+  const icons: [ComponentType<SvgProps>, string][] = [
```

Comment:
neat!

Reply:
realized i could have made this more clear with
```
  const icons: [Icon: ComponentType<SvgProps>, key: string][] = [
```
updated!